src/lib/components/actions.svelte

```diff
@@ -127,6 +128,32 @@
 		window.location.href = `${window.location.pathname}?gist=${gistURL}`;
 	};
 
+	const textEncode = (str) => {
+		if (window.TextEncoder) {
+			return new TextEncoder('utf-8').encode(str);
+		}
+		let utf8 = unescape(encodeURIComponent(str));
+		let result = new Uint8Array(utf8.length);
+		for (let i = 0; i < utf8.length; i++) {
+			result[i] = utf8.charCodeAt(i);
+		}
+		return result;
+	};
+
+	const onKrokiClick = () => {
+		const stateCopy = JSON.parse(JSON.stringify($codeStore));
+		const krokiCode = getKrokiCode(stateCopy.code);
```

Comment:
I don't think you have to make a copy of state here.

Reply:
Fixed c59351c